src/graphql/getPurchasedGames.ts

```diff
@@ -0,0 +1,68 @@
+import { AuthorizationPayload } from "../models";
+import { GetPurchasedGamesResponse } from "../models/purchased-games-response.model";
+import { call } from "../utils/call";
+import { GRAPHQL_BASE_URL } from "./GRAPHQL_BASE_URL";
+
+export type GetPurchasedGamesOptions = {
+  platform?: Array<"ps4" | "ps5">;
+  size?: number;
+  start?: number;
+  sortBy?: string;
+  sortDirection?: "asc" | "desc";
+  isActive?: boolean;
+  subscriptionService?: string;
+};
+
+const defaultOptions: GetPurchasedGamesOptions = {
+  platform: ["ps4", "ps5"],
+  size: 500,
+  start: 0,
+  sortBy: "ACTIVE_DATE",
+  sortDirection: "desc",
+  isActive: true,
+  subscriptionService: "NONE"
+};
+
+/**
+ * A call to this function will retrieve the list of games for the logged in user.
+ * If the user cannot be found (either due to non-existence or privacy settings),
+ * an error will be thrown.
+ *
+ * @param authorization An object containing your access token, typically retrieved with `exchangeCodeForAccessToken()`.
+ * @param options.platform An array containing the platforms to filter by. Defaults to `["ps4", "ps5"]`.
+ * @param options.size The number of results to return. Defaults to `500`.
+ * @param options.start The offset to start from. Defaults to `0`.
+ * @param options.sortBy The field to sort by. Defaults to `"ACTIVE_DATE"`.
+ * @param options.sortDirection The direction to sort by. Defaults to `"desc"`.
+ * @param options.isActive Whether to return active games only. Defaults to `true`.
+ * @param options.subscriptionService The subscription service to filter by. Defaults to `"NONE"`.
+ */
+export const getPurchasedGames = async (
+  authorization: AuthorizationPayload,
+  options: Partial<GetPurchasedGamesOptions> = defaultOptions
+): Promise<any> => {
```

Comment:
This strips all type metadata from the return value. In other words, while the value is typed _within_ psn-api, it will be untyped for anyone consuming the library. You can see this in the playground:

```ts
  const purchasedGames = await getPurchasedGames(authorization);

  console.log(purchasedGames.data); // the type of `data` is `any` 💀
```